deps/verifier/src/tdx/eventlog.rs

```diff
@@ -55,6 +61,30 @@ fn read_string(raw_bytes: &[u8]) -> Result<String, std::string::FromUtf16Error>
     String::from_utf16(utf16_string.as_ref())
 }
 
+fn is_qemu_direct_boot(desc: &[u8]) -> bool {
+    let mut pos = UEFI_IMAGE_LOAD_EVENT_OFFSET;
+
+    // UEFI Image Load Event contains a Device Path
+    if u64::from_le_bytes(desc[pos..pos + 8].try_into().unwrap_or_default()) == 0 {
```

Comment:
Let's add some logic to judge if `desc.len()` is `>= UEFI_IMAGE_LOAD_EVENT_OFFSET+8+2+2+16` and then index it to avoid panic

Reply:
You might consider casting the entire thing to a packed struct and then accessing the various fields. Current approach is ok, though.

Reply:
Maybe I didn't look closely enough, but it seems like all the slices are taking hardcoded lengths.

Reply:
the (Vendor) GUID part is valid for the vendor "media device path" EDK2 uses for Qemu. There are other path types/sub-types per UEFI spec "Type and Sub-Type define type of data. Size of data is included in Length".